rustls-platform-verifier/Cargo.toml

```diff
@@ -36,7 +36,7 @@ jni = { version = "0.19", default-features = false, optional = true } # Only use
 once_cell = "1.9"
 paste = { version = "1.0", default-features = false, optional = true } # Only used when `ffi-testing` feature is enabled
 
-[target.'cfg(all(unix, not(target_os = "android"), not(target_os = "macos"), not(target_os = "ios"), not(target_os = "tvos"), not(target_arch = "wasm32")))'.dependencies]
+[target.'cfg(all(unix, not(target_os = "android"), not(target_vendor = "apple"), not(target_arch = "wasm32")))'.dependencies]
```

Comment:
Nit: feels weird to have a first commit that adds `target_os = "ios"` only to then have a commit that replace the Apple `target_os` attributes with `target_vendor = "apple"`, maybe we should do it the other way around and replace with `target_vendor` first, then add specific `ios` stuff where needed?

Reply:
I originally had this all in one commit but decided to split it out 😅. I can flip the order since you think its better that way.

Reply:
Yeah, I usually like to minimize within-PR churn between commits.

Reply:
I've force pushed a re-arrangement of the commits so the Catalyst testing builds ontop of the `target_vendor = "apple"` base instead. No rush to re-review though since it will probably be next week when GitHub bumps their Rust versions.